presto-main/src/main/java/com/facebook/presto/operator/aggregation/GenericAccumulatorFactory.java

```diff
@@ -168,23 +176,57 @@ public GroupedAccumulator createGroupedAccumulator()
             }
 
             accumulator = new DistinctingGroupedAccumulator(accumulator, argumentTypes, inputChannels, maskChannel, session, joinCompiler);
+            if (!hasOrderBy()) {
+                accumulator = new SpillableFinalOnlyGroupedAccumulator(sourceTypes, (FinalOnlyGroupedAccumulator) accumulator);
```

Comment:
Show we create spillable accumulator based on `isSpillEnabled(session)`?

Reply:
Did a few interface changes to accomplish this. Feel free to comment.